tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/InitialMagneticFields/Test_Poloidal.cpp

```diff
@@ -44,10 +53,11 @@ struct PoloidalProxy : Poloidal {
 SPECTRE_TEST_CASE(
     "Unit.PointwiseFunctions.AnalyticData.GrMhd.InitialMagneticFields.Poloidal",
     "[Unit][PointwiseFunctions]") {
-  register_derived_classes_with_charm<InitialMagneticField>();
+  register_factory_classes_with_charm<Metavariables>();
```

Comment:
[optional] I think the only change you need in this file is making this line `register_classes_with_charm<InitialMagneticFields::Poloidal>();`.  The version you've written isn't wrong, but it's a lot more verbose than necessary.

Same for `Test_Toroidal.cpp`.